File: app/src/main/java/com/example/pizzeria/dao/DAOUsuarios.java
package com.example.pizzeria.dao;

import com.example.pizzeria.clases.Usuario;

import java.util.ArrayList;
import java.util.List;

public class DAOUsuarios {

    private static DAOUsuarios instance = null;
    private List<Usuario> usuarios;

    private DAOUsuarios() {
        usuarios = new ArrayList();
        usuarios.add(new Usuario("Adrian_oliva", "1234Abcd"));
    }

    public static DAOUsuarios getInstance() {
        if (instance == null) instance = new DAOUsuarios();

        return instance;
    }

    public void resetearUsuarios() {
        usuarios = new ArrayList();
    }

    public List<Usuario> getUsuarios() {
        return usuarios;
    }

    public Usuario getUsuario(Usuario usuario) {
        Usuario usuarioDevuelto = null;

        for (Usuario usuarioAct : usuarios) if (usuarioAct.equals(usuario)) usuarioDevuelto = usuarioAct;

        return usuarioDevuelto;
    }

    public Boolean insertarUsuario(Usuario usuario) {
        Boolean insertado = true;

        if (usuarios.contains(usuario)) insertado = false;
        else usuarios.add(usuario);

        return insertado;
    }

}

File: app/src/main/java/com/example/pizzeria/dao/DAOUsuarios.kt
package com.example.pizzeria.dao

import com.example.pizzeria.clases.Usuario

class DAOUsuarios private constructor(private var usuarios: List<Usuario> = ArrayList()) {

    companion object {
        private var instance: DAOUsuarios? = null
        fun getInstance(): DAOUsuarios? {
            if (instance == null) instance = DAOUsuarios()

            return instance
        }
    }

    init {
        usuarios += Usuario("Adrian_oliva", "1234Abcd")
    }

    fun resetearUsuarios() {
        usuarios = ArrayList()
    }

    fun getUsuarios(): List<Usuario> {
        return usuarios
    }

    fun getUsuario(usuario: Usuario): Usuario? {
        var usuarioDevuelto: Usuario? = null

        for (usuarioAct: Usuario in usuarios) if (usuarioAct.equals(usuario)) usuarioDevuelto = usuarioAct

        return usuarioDevuelto
    }

    fun insertarUsuario(usuario: Usuario): Boolean {
        var insertado = true

        if (usuarios.contains(usuario)) insertado = false
        else usuarios += usuario

        return insertado
    }

}
